custom/icds/messaging/indicators.py

```diff
@@ -269,6 +269,31 @@ def get_messages(self, language_code=None):
         return []
 
 
+class AWWVHNDSurveyIndicator(AWWIndicator):
+    template = 'aww_vhnd_survey.txt'
+    slug = 'phase2_aww_1'
+
+    def __init__(self, domain, user):
+        super(AWWVHNDSurveyIndicator, self).__init__(domain, user)
+
+        self.forms = get_last_form_submissions_by_user(
+            domain, [self.user.get_id], xmlns=VHND_SURVEY_XMLNS
+        )
+
+    def get_messages(self, language_code=None):
+        def convert_to_date(date):
+            return string_to_datetime(date).date() if date else None
+
+        now_date = self.now.date()
+        for forms in self.forms.values():
+            vhnd_date = convert_to_date(forms[0]['form']['vhsnd_date_past_month'])
+            if (now_date - vhnd_date).days < 37:
```

Comment:
i realize the LS version of this one was also behaving this way, but could we use `.get('vhsnd_date_past_month')` to more gracefully handle when the key doesn't exist in the form result for this data element, and also check if `vhnd_date` is not None before doing the subtraction?

and since they're small changes if we could also update the LS indicator to add those two checks that would be great as well.

Reply:
Good idea. https://github.com/dimagi/commcare-hq/pull/21504/commits/cdc9883404c19feeeea71e4a068bcb9459917876